index.bs

```diff
@@ -632,19 +632,281 @@ Issue(12): Propose control protocol for Remote
+authenticate it.
+
+This can be addressed through a combination of techniques.  The first is flagging:
+
+* Flag advertised displays whose friendly name, public key fingerprint, or
+    IP:port collide with an already trusted display.
+* Flag already-trusted displays whose metadata has changed.
+* Flag controllers or receivers that fail the authentication challenge a certain
+    number of times.
+  
+Flagging means that the user is notified, or in some cases they are required to
+re-authenticate to the presentation display to verify its identity.
+  
+The second is through management of the shared secret during mutual
+authentication:
+
+* Rotate the shared secret to prevent brute force attacks.
+* Use an increasing backoff to repond to authentication challenges, also to
+    prevent brute force attacks.
+* Use a cryptographically sound source of entropy to generate the shared secret.
+* Require the end user to manually type the shared secret - shown only the
+    display - to prevent the user from blindly clicking through this step.
+
+The active attacker may also attempt to disrupt data exchanged over the QUIC
+connection by injecting or modifying traffic.  These attacks should be mitigated
+by a correct implementation of TLS 1.3.
+
+Issue(130): [Security] Review attack and mitigation considerations for TLS 1.3
+
+### Remote active network attackers ### {#remote-active-mitigations}
+
+Unfortunately, we cannot rely on network devices to fully protect Open Screen
+controllers and receivers from traffic from the broader Internet.  Open Screen
+Protocol implementations that are only intended to work on the LAN should filter
+packets from non-local IP addresses.  Implementations can also use the ARP cache
+to detect attempts to spoof local network IP addresses.
+
+Issue(131): [Security] Mitigations for remote network attackers
+
+### Denial of service ### {#denial-of-service-mitigations}
+
+It will be difficult to completely prevent denial service of attacks that
+originate on the user's local area network.  Open Screen Protocol
+implementations can refuse new connections, rate limit the rate of messages from
+existing connections, or limit the number of mDNS records cached in an attempt
+to allow existing activities to continue in spite of such an attack.
+
+### Malicious input ### {#malicious-input-mitigations}
+
+Open Screen Protocol implementations should be robust against malicious input
+that attempts to compromise the target device by exploiting parsing
+vulnerabilities.
+
+CBOR is intended to be less vulnerable to such attacks relative to alternatives
+like JSON and XML.  Still, implementations should be thoroughly tested using
+approaches like [fuzz testing](https://en.wikipedia.org/wiki/Fuzzing).
+
+Where possible, Open Screen Protocol implementations (including the content
+rendering components) should use defense-in-depth techniques like <a
+href="https://en.wikipedia.org/wiki/Sandbox_(computer_security)">sandboxing</a>
```

Comment:
Are markdown links or HTML <a> tags preferred for hyperlinks?  Both are used within a few lines of each other, which seems odd.

Reply:
This is actually to work around a bug in Bikeshed :-)
See https://github.com/tabatkins/bikeshed/issues/1419